code/obj/item/clothing/helmets.dm

```diff
@@ -384,6 +384,16 @@
 			light_dir.update(0)
 		return
 
+	attackby(var/obj/item/T, mob/user as mob)
+		if(istype(T, /obj/item/device/prox_sensor))
+			boutput(user,  "You attach the proximity sensor to the hard hat. Now you need to add a robot arm.")
+			new /obj/item/digbotassembly(get_turf(user))
```

Comment:
why not get_turf(src)? This kinda awkwardly moves it to your tile if you're mass producing digbot assemblies on the floor